docs/cookbook/recipe_creating_a_locale_switcher.rst

```diff
@@ -0,0 +1,161 @@
+Creating a locale switcher
+==========================
+
+This is a full working example of creating a locale switcher.
+
+.. note::
+    This article assumes you are using Symfony 4, Flex, and autowiring.
+
+Prerequisites
+-------------
+
+Install following Sonata dependencies:
+
+* `Sonata Easy-Extends Bundle`_
+* `Sonata Intl Bundle`_
+
+Create ``LocaleSubscriber`` and ``UserLocaleSubscriber`` as explained `here`_.
+
+.. note::
+    Sonata Intl Bundle is optional, but will allow list of locales to be shown as list of languages.
+
+Handle locale switch
+--------------------
+
+Create a controller to handle locale switch in ``src/Controller/LocaleController.php``:
+
+.. code-block:: php
+
+    <?php
+
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\Routing\Annotation\Route;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\Session\SessionInterface;
+    use Symfony\Component\HttpFoundation\RedirectResponse;
+
+    class LocaleController extends AbstractController
+    {
+        /**
+        * @Route("/locale/{locale}", name="locale")
+        */
+        public function index(Request $request, SessionInterface $session, $locale)
+        {
+            $session->set('_locale', $locale);
+            $referer = $request->headers->get('referer');
+            return new RedirectResponse($referer ?: '/');
+        }
+    }
+
+Extend SonataAdminBundle
+------------------------
+
+Open a command console, enter your project directory and execute the
+following command:
+
+.. code-block:: bash
+
+    $ bin/console sonata:easy-extends:generate SonataAdminBundle --dest=src --namespace_prefix=App
```

Comment:
I think this part is not really needed, you can just create a template in your own directory.

Reply:
OK, so it means using EasyExtends is not recommended with sf4?

Reply:
not really, most of sonata packages have flex recipes, also in this case I think easy extends is not really needed and you can create that template in your own directory

Reply:
Thanks for your advise, I removed use of easy extends in last commit.
Before to resolve, could you tell me in few words in wich case it makes sense to use easy extends with sf4 ?

Reply:
@kunicmarko20 If you succeed to make me understand the purpose of easy extends bundle, I promess to create a PR on easyextends bundle documentation ;)

Reply:
So doc of bundles listed on https://github.com/symfony/recipes-contrib/tree/master/sonata-project should be updated to tell people not to use easy-extends?

Reply:
@kunicmarko20 so if I understand correctly, with classification bundle by exemple, as a recipe exist it would be nice to create a PR on doc that simply replace installation process by `composer require sonata-project/classification-bundle`, or maybe to split installation chapter between automatic and manual ? Is it so simple?